inc/search.class.php

```diff
@@ -4846,7 +4846,8 @@ static function addLeftJoin($itemtype, $ref_table, array &$already_link_tables,
       }
 
       // Multiple link possibilies case
-      if (!empty($linkfield) && ($linkfield != getForeignKeyFieldForTable($new_table))) {
+      $ref_fk = getForeignKeyFieldForTable($new_table);
+      if (!empty($linkfield) && ($linkfield != $ref_fk) && strpos($linkfield, $ref_fk . '_') !== 0) {
```

Comment:
Could be `if (!empty($linkfield) && getTableNameForForeignKeyField($linkfield) != $new_table) {`

Reply:
Tests failed with this change. I reverted it.

Reply:
Tests were failing on change made to line 4871 (my first suggestion which was wrong).